From e0a337a2f7331533c916eb53dbc9b046331ab87f Mon Sep 17 00:00:00 2001
From: Ye ShanShan <yeshanshan@uniontech.com>
Date: Tue, 11 Mar 2025 14:14:19 +0800
Subject: [PATCH] feat: export the symbol of qtkey and qttouch protocol

Support to extent qtkey and qttouch by caller,
It's to assist linuxdeepin/dde-shell#1058 in solving
Num can't work in dde-shell.

Change-Id: I5d9dc68fa139ec590f295b3638b85718c53f177d
---
 src/compositor/extensions/qwlqtkey_p.h   | 2 +-
 src/compositor/extensions/qwlqttouch_p.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/compositor/extensions/qwlqtkey_p.h b/src/compositor/extensions/qwlqtkey_p.h
index e7cff960..6738b1bd 100644
--- a/src/compositor/extensions/qwlqtkey_p.h
+++ b/src/compositor/extensions/qwlqtkey_p.h
@@ -29,7 +29,7 @@ class QKeyEvent;
 
 namespace QtWayland {
 
-class QtKeyExtensionGlobal : public QWaylandCompositorExtensionTemplate<QtKeyExtensionGlobal>, public QtWaylandServer::zqt_key_v1
+class Q_WAYLANDCOMPOSITOR_EXPORT QtKeyExtensionGlobal : public QWaylandCompositorExtensionTemplate<QtKeyExtensionGlobal>, public QtWaylandServer::zqt_key_v1
 {
     Q_OBJECT
 public:
diff --git a/src/compositor/extensions/qwlqttouch_p.h b/src/compositor/extensions/qwlqttouch_p.h
index 2d96da7f..52e0733c 100644
--- a/src/compositor/extensions/qwlqttouch_p.h
+++ b/src/compositor/extensions/qwlqttouch_p.h
@@ -30,7 +30,7 @@ class QWaylandView;
 
 namespace QtWayland {
 
-class TouchExtensionGlobal : public QWaylandCompositorExtensionTemplate<TouchExtensionGlobal>, public QtWaylandServer::qt_touch_extension
+class Q_WAYLANDCOMPOSITOR_EXPORT TouchExtensionGlobal : public QWaylandCompositorExtensionTemplate<TouchExtensionGlobal>, public QtWaylandServer::qt_touch_extension
 {
     Q_OBJECT
     Q_PROPERTY(BehaviorFlags behaviorFlags READ behaviorFlags WRITE setBehviorFlags NOTIFY behaviorFlagsChanged)
-- 
2.45.2

